### PR TITLE
Add simnet guide to advanced section.

### LINKED ISSUE #188: Mainnet vs Simnet vs Testnet
The difference between these, what they should be used for. How to use them.

I think the only thing we already have on this topic is [the different type of addresses]( https://docs.decred.org/faq/wallets-and-seeds/#9-what-are-the-different-types-of-addresses)

[Here](https://gist.github.com/davecgh/2992ed85d41307e794f6) is a good resource explaining simnet for btcd, which is mostly analogous for dcrd.

Could mention testnet faucet here as well - the only place we currently mention it is in the Testnet voting guide, and I assume we will be replacing that soon in favour of a Mainnet voting guide

### PR DESCRIPTION
**Pull request by jholdstock**
Closes #188.

Info taken from https://gist.github.com/davecgh/2992ed85d41307e794f6#file-using_btcd_simnet-md

### CODE REVIEW

File: de_docs/advanced/simnet.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# **Simnet**
+
+---
+
+## **Overview**
+
+When developing Decred applications or testing potential changes, it is often extremely useful to have a test network where transactions are actually mined into blocks and difficulty levels are low enough to generate blocks as needed.
```
Comment (jcvernaleo) [inline de_docs/advanced/simnet.md]: Not sure what the part 'transactions are actually mined into blocks' even means.  Probably can just drop that bit.

File: de_docs/advanced/simnet.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    `$ dcrwallet --simnet --username=<username> --password=<password>`
+
+* Create a new simnet Decred address:
+
+    `$ dcrctl --simnet --wallet --rpcuser=<username> --rpcpass=<password> getnewaddress`
+
+* Stop the initial dcrd process and restart it with the mining address set to the output from the previous command:
+
+    `$ dcrd --simnet --rpcuser=<username> --rpcpass=<password> --miningaddr=S....`
```
Comment (jcvernaleo) [inline de_docs/advanced/simnet.md]: Shouldn't the mining address also be in the form `<S....?>` ?

**Comment by jholdstock** [comment]
Review comments addressed

**Comment by jcvernaleo** [comment]
Looks all good now.

**Pull request merged**